engine/src/main/java/org/terasology/physics/engine/VoxelLiquidWorldSystem.java

```diff
@@ -146,20 +146,15 @@ public void onNewChunk(OnChunkLoaded chunkAvailable, EntityRef worldEntity) {
         ByteBuffer buffer =
             ByteBuffer.allocateDirect(2 * (ChunkConstants.SIZE_X * ChunkConstants.SIZE_Y * ChunkConstants.SIZE_Z));
         buffer.order(ByteOrder.nativeOrder());
-        short[] entries = new short[ChunkConstants.SIZE_X * ChunkConstants.SIZE_Y * ChunkConstants.SIZE_Z];
-        for (int x = 0; x < ChunkConstants.SIZE_X; x++) {
-            for (int y = 0; y < ChunkConstants.SIZE_Y; y++) {
-                for (int z = 0; z < ChunkConstants.SIZE_Z; z++) {
-                    int index = (z * ChunkConstants.SIZE_X * ChunkConstants.SIZE_Y) + (x * ChunkConstants.SIZE_Y) + y;
+        for (int z = 0; z < ChunkConstants.SIZE_Z; z++) {
+            for (int x = 0; x < ChunkConstants.SIZE_X; x++) {
+                for (int y = 0; y < ChunkConstants.SIZE_Y; y++) {
```

Comment:
What is the reason the order here changed from x, y, z to **z, x, y**?

Reply:
This is how the data are organized in the Buffer (see. index variable on removed line 153 which has Y as 1+ increment)